src/main/java/Duke.java

```diff
@@ -1,10 +1,375 @@
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Please input a valid index.");
+                }
+                break;
+            default:
+                System.out.println("Invalid! Please try again.");
+            }
+            try {
+                userInput = sc.nextLine();
+                commandInput = identifyUserInput(userInput)[0];
+                checkedCommandInput = commandInputError(commandInput);
+            }catch (IncorrectCommandInput e){
+                System.out.println("Invalid Command! Please try again");
+            }
+        }
+        printGoodbyeMessage();
+    }
+
+    private static void deleteTaskFromList(TaskList taskList, String userInput){
+        String userInputIndex = identifyUserInput(userInput)[1];
+        int index = Integer.parseInt(userInputIndex);
+        taskList.deleteTask(index);
+        System.out.println("Item " + userInputIndex + " has been deleted");
+        System.out.println("Here is the new list: ");
+        listTask(taskList);
+        createNewOutputFile(taskList);
+    }
+
+    /**
+     * Check if there are existing saved files.
+     * If not will create a new one.
+     * @param taskList
+     */
+    public static void checkExistingOutputFile(TaskList taskList){
+        try{
+            Path path = Paths.get("duke.txt");
+            BufferedReader bufferedReader = new BufferedReader(
+                    new FileReader(path.toAbsolutePath().toString()));
+            String outputFileLineText = "";
+            while((outputFileLineText = bufferedReader.readLine()) != null){
+                String command = outputFileLineText.substring(0, 3);
+                switch(command){
+                case "[T]":
+                    addToDoTaskFromSavedList(taskList, outputFileLineText);
+                    break;
+                case "[D]":
+                    addDeadlineTaskFromSavedList(taskList,outputFileLineText);
+                    break;
+                case "[E]":
+                    addEventTaskFromSavedList(taskList,outputFileLineText);
+                    break;
+                }
+            }
+            bufferedReader.close();
+            listTask(taskList);
+        } catch (FileNotFoundException e){
+            System.out.println("There is no such file, a new file named duke.txt will be created");
+
+        } catch(IOException e){
+            System.out.println("There is no text in the file");
+            System.out.println(e);
+        }
+    }
+
+    /**
+     * Adds todo task from saved list to program list.
+     * @param taskList
+     * @param userInput
+     */
+    private static void addToDoTaskFromSavedList(TaskList taskList, String userInput) {
+        boolean completed = false;
+        String taskDescription = "";
+        if(userInput.contains("[X]")){
+            completed = true;
+            //taskDescription = userInput.split(" ", 3)[2];
```

Comment:
Where there is commented code, you could remove it to improve code readability. You have several commented lines of code spread out in this file. Use[ comments](https://se-education.org/guides/conventions/java/basic.html#comments) to describe the logic of code instead.